File: backend/app/infrastructure/database/models/device_type.py
from sqlalchemy import Column, Integer, String

from app.db.database import Base


class DeviceType(Base):
    """
    SQLAlchemy model for device types.
    """

    __tablename__ = "device_types"

    id = Column(Integer, primary_key=True, index=True)

    name = Column(String, nullable=False, unique=True)
    company = Column(String, nullable=True)
    output_spec = Column(String, nullable=True)
    adapter_name = Column(String, nullable=True)

File: frontend/src/types/deviceType.ts
export interface DeviceType {
  id: number;
  name: string;
  company: string;
  output_spec?: string;
  adapter_name?: string;
  created_at: string;
  updated_at: string;
}

export interface DeviceTypePayload {
  name: string;
  company: string;
  output_spec?: string;
  adapter_name?: string;
}